custom_model_runner/datarobot_drum/drum/typeschema_validation.py

```diff
@@ -187,6 +187,26 @@ def _get_mapping(field: Fields, values: List[Values]) -> Map:
     return Map({"field": Enum(str(field)), "condition": conditions, "value": value_enum})
 
 
+def check_is_mm(data):
+    if data.shape[1] == 1 and "%%MatrixMarket matrix coordinate" in data.columns[0]:
+        return True
+    else:
+        return False
+
+
+def check_csr_sparse(data):
+    try:
+        return data.format == "csr"
+    except AttributeError:
+        return data.dtypes.apply(pd.api.types.is_sparse).any()
```

Comment:
Where and why do we load sparse data into a pandas dataframe?

Reply:
if the transform returns it as sparse, when it is loaded to test in the type schema.

Reply:
But where do we put a CSR matrix inside of a pandas dataframe? I feel like we should just be passing the CSR itself